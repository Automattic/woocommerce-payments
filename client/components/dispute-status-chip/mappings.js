/** @format **/
/* eslint-disable camelcase */

/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';

export default {
	warning_needs_response: {
		type: 'primary',
		message: __( 'Inquiry: Needs response', 'woocommerce-payments' ),
	},
	warning_under_review: {
		type: 'light',
		message: __( 'Inquiry: Under review', 'woocommerce-payments' ),
	},
	warning_closed: {
		type: 'light',
		message: __( 'Inquiry: Closed', 'woocommerce-payments' ),
	},
	needs_response: {
		type: 'primary',
		message: __( 'Needs response', 'woocommerce-payments' ),
	},
	under_review: {
		type: 'light',
		message: __( 'Under review', 'woocommerce-payments' ),
	},
	charge_refunded: {
		type: 'light',
		message: __( 'Charge refunded', 'woocommerce-payments' ),
	},
	won: {
		type: 'light',
		message: __( 'Won', 'woocommerce-payments' ),
	},
	lost: {
		type: 'light',
		message: __( 'Lost', 'woocommerce-payments' ),
	},
};
