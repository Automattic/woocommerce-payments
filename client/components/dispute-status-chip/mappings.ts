/** @format **/

/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import type { ChipType } from '../chip';

const status: {
	[ key: string ]: {
		type: ChipType;
		message: string;
	};
} = {
	warning_needs_response: {
		type: 'warning',
		message: __( 'Inquiry: Needs response', 'woocommerce-payments' ),
	},
	warning_under_review: {
		type: 'primary',
		message: __( 'Inquiry: Under review', 'woocommerce-payments' ),
	},
	warning_closed: {
		type: 'light',
		message: __( 'Inquiry: Closed', 'woocommerce-payments' ),
	},
	needs_response: {
		type: 'warning',
		message: __( 'Needs response', 'woocommerce-payments' ),
	},
	under_review: {
		type: 'primary',
		message: __( 'Under review', 'woocommerce-payments' ),
	},
	charge_refunded: {
		type: 'light',
		message: __( 'Charge refunded', 'woocommerce-payments' ),
	},
	won: {
		type: 'success',
		message: __( 'Won', 'woocommerce-payments' ),
	},
	lost: {
		type: 'light',
		message: __( 'Lost', 'woocommerce-payments' ),
	},
};

export default status;
