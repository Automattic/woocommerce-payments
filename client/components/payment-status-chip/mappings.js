/** @format **/
/* eslint-disable camelcase */

/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';

/* TODO: implement other payment statuses (SCA and authorizations) */
export default {
	refunded_partial: {
		type: 'light',
		message: __( 'Partial refund', 'woocommerce-payments' ),
	},
	refunded_full: {
		type: 'light',
		message: __( 'Refunded', 'woocommerce-payments' ),
	},
	paid: {
		type: 'light',
		message: __( 'Paid', 'woocommerce-payments' ),
	},
	authorized: {
		type: 'primary',
		message: __( 'Payment authorized', 'woocommerce-payments' ),
	},
	failed: {
		type: 'alert',
		message: __( 'Payment failed', 'woocommerce-payments' ),
	},
	blocked: {
		type: 'alert',
		message: __( 'Payment blocked', 'woocommerce-payments' ),
	},
	disputed_needs_response: {
		type: 'primary',
		message: __( 'Disputed: Needs response', 'woocommerce-payments' ),
	},
	disputed_under_review: {
		type: 'light',
		message: __( 'Disputed: In review', 'woocommerce-payments' ),
	},
	disputed_won: {
		type: 'light',
		message: __( 'Disputed: Won', 'woocommerce-payments' ),
	},
	disputed_lost: {
		type: 'light',
		message: __( 'Disputed: Lost', 'woocommerce-payments' ),
	},
	disputed_closed: {
		type: 'light',
		message: __( 'Disputed: Inquiry closed', 'woocommerce-payments' ),
	},
};
