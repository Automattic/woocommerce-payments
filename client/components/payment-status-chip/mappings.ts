/** @format **/

/**
 * External dependencies
 */
import { __, sprintf } from '@wordpress/i18n';
import { objectEntries } from 'ts-extras';

/**
 * Internal dependencies
 */
import disputeStatuses from 'components/dispute-status-chip/mappings';
import type { ChipType } from '../chip';

type DisputeStatus = keyof typeof disputeStatuses;
// prettier-ignore
// eslint-disable-next-line
type DisputedStatus = `disputed_${ DisputeStatus }`;

const formattedDisputeStatuses = objectEntries( disputeStatuses ).reduce(
	( statuses, [ status, mapping ] ) => {
		statuses[ `disputed_${ status }` as const ] = {
			type: mapping.type,
			message: status.startsWith( 'warning_' )
				? mapping.message
				: sprintf(
						/** translators: %s dispute status, e.g. Won, Lost, Under review, etc. */
						__( 'Disputed: %s', 'woocommerce-payments' ),
						mapping.message
				  ),
		};
		return statuses;
	},
	{} as Record<
		DisputedStatus,
		{ type: ChipType; message: string }
	>
);

/* TODO: implement other payment statuses (SCA and authorizations) */
export default {
	refunded_partial: {
		type: 'light',
		message: __( 'Partial refund', 'woocommerce-payments' ),
	},
	refunded_full: {
		type: 'light',
		message: __( 'Refunded', 'woocommerce-payments' ),
	},
	paid: {
		type: 'light',
		message: __( 'Paid', 'woocommerce-payments' ),
	},
	authorized: {
		type: 'primary',
		message: __( 'Payment authorized', 'woocommerce-payments' ),
	},
	refund_failed: {
		type: 'alert',
		message: __( 'Refund failure', 'woocommerce-payments' ),
	},
	failed: {
		type: 'alert',
		message: __( 'Payment failed', 'woocommerce-payments' ),
	},
	blocked: {
		type: 'alert',
		message: __( 'Payment blocked', 'woocommerce-payments' ),
	},
	fraud_outcome_review: {
		type: 'warning',
		message: __( 'Needs review', 'woocommerce-payments' ),
	},
	fraud_outcome_block: {
		type: 'alert',
		message: __( 'Payment blocked', 'woocommerce-payments' ),
	},
	...formattedDisputeStatuses,
} as const;
