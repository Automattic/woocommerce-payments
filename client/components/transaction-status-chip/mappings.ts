/** @format **/

/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';

const transactionStatusMapping = {
	allow: {
		type: 'primary',
		message: __( 'Succeeded', 'woocommerce-payments' ),
	},
	review: {
		type: 'warning',
		message: __( 'Needs review', 'woocommerce-payments' ),
	},
	block: {
		type: 'alert',
		message: __( 'Blocked', 'woocommerce-payments' ),
	},
};

export type TransactionStatus = keyof typeof transactionStatusMapping;

export default transactionStatusMapping;
