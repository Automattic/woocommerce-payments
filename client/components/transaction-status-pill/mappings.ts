/** @format **/

/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import type { PillType } from 'components/pill';

export type TransactionStatus = 'allow' | 'review' | 'block';
type TransactionStatusMapping = Record<
	TransactionStatus,
	{
		type: PillType;
		message: string;
	}
>;
const transactionStatusMapping: TransactionStatusMapping = {
	allow: {
		type: 'success',
		message: __( 'Succeeded', 'woocommerce-payments' ),
	},
	review: {
		type: 'alert',
		message: __( 'Needs review', 'woocommerce-payments' ),
	},
	block: {
		type: 'danger',
		message: __( 'Payment blocked', 'woocommerce-payments' ),
	},
};

export default transactionStatusMapping;
