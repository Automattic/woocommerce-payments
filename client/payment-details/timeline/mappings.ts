/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import {
	Outcomes,
	Rules,
} from '../../settings/fraud-protection/advanced-settings/constants';

export const fraudOutcomeRulesetMapping = {
	[ Outcomes.REVIEW ]: {
		[ Rules.RULE_ADDRESS_MISMATCH ]: __(
			'Place in review if the shipping address country differs from the billing address country',
			'woocommerce-payments'
		),
		[ Rules.RULE_INTERNATIONAL_IP_ADDRESS ]: __(
			'Place in review if the country resolved from customer IP is not listed in your selling countries',
			'woocommerce-payments'
		),
		[ Rules.RULE_IP_ADDRESS_MISMATCH ]: __(
			'Place in review if the order originates from a country that`s different from the shipping address country',
			'woocommerce-payments'
		),
		[ Rules.RULE_ORDER_ITEMS_THRESHOLD ]: __(
			'Place in review if the items count is not in your defined range',
			'woocommerce-payments'
		),
		[ Rules.RULE_PURCHASE_PRICE_THRESHOLD ]: __(
			'Place in review if the purchase price is not in your defined range',
			'woocommerce-payments'
		),
	},
	[ Outcomes.BLOCK ]: {
		[ Rules.RULE_ADDRESS_MISMATCH ]: __(
			'Block if the shipping address differs from the billing address',
			'woocommerce-payments'
		),
		[ Rules.RULE_INTERNATIONAL_IP_ADDRESS ]: __(
			'Block if the country resolved from customer IP is not listed in your selling countries',
			'woocommerce-payments'
		),
		[ Rules.RULE_IP_ADDRESS_MISMATCH ]: __(
			'Block if the order originates from a country that`s different from the shipping address country',
			'woocommerce-payments'
		),
		[ Rules.RULE_ORDER_ITEMS_THRESHOLD ]: __(
			'Block if the items count is not in your defined range',
			'woocommerce-payments'
		),
		[ Rules.RULE_PURCHASE_PRICE_THRESHOLD ]: __(
			'Block if the purchase price is not in your defined range',
			'woocommerce-payments'
		),
	},
};
