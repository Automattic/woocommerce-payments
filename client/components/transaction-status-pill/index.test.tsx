/**
 * External dependencies
 */
import React from 'react';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import TransactionStatusPill from '.';
import { TransactionStatus } from './mappings';

describe( 'Transaction status pill', () => {
	[ 'allow', 'review', 'block', 'unknown' ].forEach( ( status: string ) => {
		it( `should render the ${ status } pill correctly`, () => {
			const { container } = render(
				<TransactionStatusPill status={ status as TransactionStatus } />
			);

			expect( container ).toMatchSnapshot();
		} );
	} );
} );
