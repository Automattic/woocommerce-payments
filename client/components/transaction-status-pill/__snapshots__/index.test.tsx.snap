// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction status pill should render the allow pill correctly 1`] = `
<div>
  <span
    class="components-truncate components-text woocommerce-pill wcpay-pill__success css-1g4vivm-View-Text-sx-Base em57xhy0"
    data-wp-c16t="true"
    data-wp-component="Text"
  >
    Succeeded
  </span>
</div>
`;

exports[`Transaction status pill should render the block pill correctly 1`] = `
<div>
  <span
    class="components-truncate components-text woocommerce-pill wcpay-pill__danger css-1g4vivm-View-Text-sx-Base em57xhy0"
    data-wp-c16t="true"
    data-wp-component="Text"
  >
    Payment blocked
  </span>
</div>
`;

exports[`Transaction status pill should render the review pill correctly 1`] = `
<div>
  <span
    class="components-truncate components-text woocommerce-pill wcpay-pill__alert css-1g4vivm-View-Text-sx-Base em57xhy0"
    data-wp-c16t="true"
    data-wp-component="Text"
  >
    Needs review
  </span>
</div>
`;

exports[`Transaction status pill should render the unknown pill correctly 1`] = `
<div>
  <span
    class="components-truncate components-text woocommerce-pill wcpay-pill__light css-1g4vivm-View-Text-sx-Base em57xhy0"
    data-wp-c16t="true"
    data-wp-component="Text"
  >
    Unknown
  </span>
</div>
`;
